components/Typography/src/MDCFontTextStyle.h

```diff
@@ -0,0 +1,35 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+/**
+ Material font text styles
+ 
+ These styles are defined in:
+ https://material.io/guidelines/style/typography.html
+ This enumeration is a set of semantic descriptions intended to describe the fonts returned by
+ + [UIFont mdc_preferredFontForMaterialTextStyle:]
+ + [UIFontDescriptor mdc_preferredFontDescriptorForMaterialTextStyle:]
+ */
+typedef NS_ENUM(NSInteger, MDCFontTextStyle) {
```

Comment:
are we planing on replacing the existing API with ones that have pass this in as an argument?

+ (nonnull UIFont *)headlineFont; et al
would get replaced by
+ (nonnull UIFont *)fontForMaterialTextStyle:(MDCFontTextStyle)style;

same for the opacities?